src/ape/contracts/base.py

```diff
@@ -106,7 +106,7 @@ def __call__(self, *args, **kwargs) -> Any:
 class ContractCallHandler:
     provider: ProviderAPI
     converter: "ConversionManager"
-    address: AddressType
+    address: AddressAPI
```

Comment:
At this point, should it be: `contract_instance: ContractInstance` instead? We actually know the link backwards is 100% always a `ContractInstance`

Reply:
I suppose. It is just weird because even though it is technically a contract instance, is not necessarily a contract. And the API we need is only in the `AddressAPI` so imo `ContractInstance` is unnecessarily specific.

But yeah, since it is always that (currently), I can make that change.

It is easier to make types more specific than it is to do the opposite, so if this ever needs to just be `AddressAPI`, going backwards could be a breaking change